Sources/AsyncHTTPClient/HTTPClient+HTTPCookie.swift

```diff
@@ -42,79 +47,71 @@ extension HTTPClient {
             self.httpOnly = false
             self.secure = false
 
-            for component in components[1...] {
-                switch self.parseComponent(component) {
-                case (nil, nil):
-                    continue
-                case ("path", .some(let value)):
-                    self.path = value
-                case ("domain", .some(let value)):
-                    self.domain = value
-                case ("expires", let value):
-                    guard let value = value else {
-                        continue
-                    }
-
-                    let formatter = DateFormatter()
-                    formatter.locale = Locale(identifier: "en_US")
-                    formatter.timeZone = TimeZone(identifier: "GMT")
+            var parsedPath: String.UTF8View.SubSequence?
+            var parsedDomain: String.UTF8View.SubSequence?
 
-                    formatter.dateFormat = "EEE, dd MMM yyyy HH:mm:ss z"
-                    if let date = formatter.date(from: value) {
-                        self.expires = date
-                        continue
+            for component in components {
+                switch component.parseCookieComponent() {
+                case ("path", let value)?:
+                    // Unlike other values, unspecified and empty paths reset to the default path.
+                    guard let value = value, value.first == UInt8(ascii: "/") else {
+                        parsedPath = nil
+                        break
                     }
-
-                    formatter.dateFormat = "EEE, dd-MMM-yy HH:mm:ss z"
-                    if let date = formatter.date(from: value) {
-                        self.expires = date
-                        continue
+                    parsedPath = value
+                case ("domain", let value)?:
+                    guard var value = value, !value.isEmpty else {
+                        break
                     }
-
-                    formatter.dateFormat = "EEE MMM d hh:mm:s yyyy"
-                    if let date = formatter.date(from: value) {
-                        self.expires = date
+                    if value.first == UInt8(ascii: ".") {
+                        value.removeFirst()
                     }
-                case ("max-age", let value):
-                    self.maxAge = value.flatMap(Int.init)
-                case ("secure", nil):
+                    guard !value.isEmpty else {
+                        parsedDomain = nil
+                        break
+                    }
+                    parsedDomain = value
+                case ("expires", let value)?:
+                    guard let value = value, let timestamp = parseCookieTime(value) else {
+                        break
+                    }
+                    self.expires_timestamp = timestamp
+                case ("max-age", let value)?:
+                    guard let value = value, let age = Int(Substring(value)) else {
```

Comment:
Fun task for the future: let's just write our own `Int.init(_ bytes: Substring.UTF8View)` initializer. It doesn't really matter, but it'd be fun and very slightly faster. Let's not do it now though, just planting seeds to keep you engaged in the future :wink:.

Reply:
Yeah, it's not too difficult. I already did a similar thing for WebURL, because the standard library's version only accepts `StringProtocol` rather than arbitrary buffers of bytes: https://github.com/karwa/swift-url/blob/1f1263f94e25eb43d8688dc668de6638e092f748/Sources/WebURL/Util/ASCII.swift#L502

But yeah, it can be a follow-up. I'll have a patch up which addresses the nits in a little bit.